### PR TITLE
[DNM] topology2: remove explicit dai_index setting for Dai object

### PR DESCRIPTION
**Pull request by yongzhi1**
For Object.Dai.XX."N", the dai_index attribute can only be intialized to its unique instance number N.

Signed-off-by: Yong Zhi <yong.zhi@intel.com>

### CODE REVIEW

**Review by ranj063 — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@yongzhi1 can you force push again to unblock CI. Thanks

**Comment by ranj063** [comment]
> @yongzhi1 can you force push again to unblock CI. Thanks

@yongzhi1 if you are going to force push, can u please fix the checkpatch errors as well?

**Comment by yongzhi1** [comment]
V2 update:
- Fix misspelled word in the commit message. Thanks!!

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@yongzhi1 looks like CI is blocked.  Can you force-push again. Thanks

**Comment by yongzhi1** [comment]
Oh, I think it's a little tricky here, the NOCODEC test failure is not a mere accident, the dai_index is not used here for the purpose of initialization because it cannot be initialized this way, but it must be visible for hw_config to build NHLT blob.

**Comment by lgirdwood** [comment]
> Oh, I think it's a little tricky here, the NOCODEC test failure which blocked CI is not a mere accident, the dai_index is not used here for the purpose of initialization because it cannot be initialized this way, but it must be visible for hw_config to build NHLT blob.

@juimonen any ideas ?

**Comment by lgirdwood** [comment]
@yongzhi1 need to wait on CI coming back to dig deeper. Marked for v2.4 so we dont forget.

**Comment by ranj063** [comment]
@yongzhi1 can we please drop this PR? I am going to introduce a change where setting the dai_index will become mandatory

**Comment by lgirdwood** [comment]
> @yongzhi1 can we please drop this PR? I am going to introduce a change where setting the dai_index will become mandatory

No objections from me if we have a better long term solution.

**Comment by kv2019i** [comment]
@ranj063  @yongzhi1  I marked this with DoNotMerge in the title now. Please update if there's a replacement.

**Comment by yongzhi1** [comment]
Ok, let me drop this PR till we

> > @yongzhi1 can we please drop this PR? I am going to introduce a change where setting the dai_index will become mandatory
> 
> No objections from me if we have a better long term solution.

Ack, let me drop this this one till long term solution, thanks everyone!!

**Pull request closed without merging**
